src/workerd/jsg/jsg.h

```diff
@@ -2053,6 +2053,26 @@ class Lock {
   bool warningsLogged;
 };
 
+class V8StackScope {
+  // One of this MUST be allocated on the stack before taking an isolate lock. It must be allocated
+  // before any handles on the stack. This MUST NOT be allocated on the heap.
+  //
+  // The reason why Isolate::Lock doesn't take care of this automatically is because it is often
+  // allocated on the heap. The purpose of V8StackScope is to capture the start of the stack
+  // range that V8 must scan when performing conservative stack-scanning garbage collection.
+public:
+  // No interface.
```

Comment:
(not necessary) I wonder if in debug mode we can compare this with stack pointer and detect when it was created on the heap.